go/samples/firebase-retrievers/main.go

```diff
@@ -43,37 +41,50 @@ func main() {
 		log.Fatal("Environment variable FIRESTORE_COLLECTION is not set")
 	}
 
-	// Initialize Firestore client
-	firestoreClient, err := firestore.NewClient(ctx, projectID, option.WithCredentialsFile(""))
+	g, err := genkit.Init(ctx)
 	if err != nil {
-		log.Fatalf("Error creating Firestore client: %v", err)
+		log.Fatal(err)
 	}
-	defer firestoreClient.Close()
 
-	// Firebase app configuration and initialization
-	firebaseApp, err := firebasev4.NewApp(ctx, nil)
+	// Mock embedder
+	embedder := &MockEmbedder{}
+	// Initialize firebase app
+	conf := &firebasev4.Config{ProjectID: projectID}
+	firebaseApp, err := firebasev4.NewApp(ctx, conf)
 	if err != nil {
-		log.Fatalf("Error initializing Firebase app: %v", err)
+		log.Fatalf("Error initializing Firebase App: %v", err)
 	}
 
-	// Firebase configuration using the initialized app
-	firebaseConfig := &firebase.FirebasePluginConfig{
-		App: firebaseApp, // Pass the pre-initialized Firebase app
+	// Initialize Firestore client
+	firestoreClient, err := firebaseApp.Firestore(ctx)
+	if err != nil {
+		log.Fatalf("Error creating Firestore client: %v", err)
 	}
+	defer firestoreClient.Close()
 
-	g, err := genkit.Init(ctx)
-	if err != nil {
-		log.Fatal(err)
+	// Firestore Retriever Configuration
+	retrieverOptions := firebase.RetrieverOptions{
+		Name:            "example-retriever",
+		Collection:      collectionName,
+		Embedder:        embedder,
+		VectorField:     "embedding",
+		ContentField:    "text",
+		MetadataFields:  []string{"metadata"},
+		Limit:           10,
+		DistanceMeasure: firestore.DistanceMeasureEuclidean,
+		VectorType:      firebase.Vector64,
+	}
+
+	f := &firebase.Firebase{
+		App:           firebaseApp,
+		RetrieverOpts: retrieverOptions,
 	}
 
 	// Initialize Firebase plugin
-	if err := firebase.Init(ctx, g, firebaseConfig); err != nil {
+	if err := f.Init(ctx, g); err != nil {
```

Comment:
The intended way for plugins to be used is to be passed to `genkit.Init()` in `WithPlugins()`. We should design around that use case.